packages/supermassive/src/__tests__/subscribe.test.ts

```diff
@@ -0,0 +1,281 @@
+import { parse, subscribe as graphQLSubscribe, isInputType } from "graphql";
+import { subscribeWithoutSchema, subscribeWithSchema } from "..";
+import schema, { typeDefs } from "../benchmarks/swapi-schema";
+import models from "../benchmarks/swapi-schema/models";
+import resolvers from "../benchmarks/swapi-schema/resolvers";
+import { addTypesToRequestDocument } from "../ast/addTypesToRequestDocument";
+import { extractImplicitTypes } from "../extractImplicitTypesRuntime";
+import { Resolvers } from "../types";
+import { specifiedScalars } from "../values";
+
+interface TestCase {
+  name: string;
+  document: string;
+  variables?: Record<string, unknown>;
+}
+
+const testCases: Array<TestCase> = [
+  {
+    name: "basic subscription with variables",
+    document: `
+  subscription emitPersons($limit: Int!) {
+    emitPersons(limit: $limit) {
+      name
+      gender
+    }
+  }
+ `,
+    variables: {
+      limit: 5,
+    },
+  },
+];
+
+const errorTestCases: Array<TestCase> = [
+  {
+    name: "subscription throw an error",
+    document: `
+  subscription emitPersons($limit: Int!, $throwError: Boolean) {
+    emitPersons(limit: $limit, throwError: $throwError) {
+      name
+      gender
+    }
+  }
+ `,
+    variables: {
+      limit: 5,
+      throwError: true,
+    },
+  },
+];
+
+describe("subscribeWithSchema", () => {
+  testCases.forEach(({ name, document, variables }: TestCase) => {
+    it(name, async () => {
+      await compareResultsForSubscribeWithSchema(document, variables);
+    });
+  });
```

Comment:
Can we switch to [`test.each` or `it.each` syntax](https://jestjs.io/docs/api#testeachtablename-fn-timeout) - IDEs support it a bit better than custom loops.